jsonrpc2/client.go

```diff
@@ -22,17 +22,33 @@ func (c *Client) NextID() int {
 }
 
 func (c *Client) Request(method string, params ...interface{}) (*Message, error) {
+	return newRequest(c.NextID(), method, params...)
+}
+
+// newNotification is a helper for creating encoded Request Messages without IDs.
+func newNotification(method string, params ...interface{}) (*Message, error) {
```

Comment:
Why not add `Notification` as a public method and to the `Requester` interface?  Or perhaps a new `Notifier` interface?

Reply:
Mostly trying to reduce the public API surface. I imagine there are two use flows for this kind of library: Either you're using it at the highest level (`Call(...)`) and never have to think about the message types, or you're using it at the lowest level and passing around message instances through codecs.

These newNotification/newRequest helpers are kind-of in-between (really they're helpers for going from low-level to high-level). If you're constructing your own message types, you probably have enough custom stuff that you're not going to use this. If you're using the high-level interface, you're not going to use this either.